Борзов Артем

qoter opened the pull request:
[Посмотреть решение](https://rawgit.com/Qoter/markup-task-4/master/index.html)

qoter committed:
Add simple markup, simple styles

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

bazhenova commented:
Самое первое, что бросается в глаза - ужасно сделаны переносы. Так лучше никогда не делать. Надо обязательно поправить

bazhenova commented:
Теперь по поводу требований. Некоторые не реализованы:

```
1) Есть, конечно, картинки и подписи к ним, иногда они занимают несколько! колонок
2) В тексте используются переносы (давай сделаем нормальные)
3) Изредка текст пишут:
В
О
Т

Т
А
К
```

bazhenova commented:
Поехала верстка при разрешении 1280х960
![image](https://cloud.githubusercontent.com/assets/12410663/19901558/408ca454-a089-11e6-8909-147bfe576a99.png)

bazhenova commented on index.html:
```diff
@@ -2,7 +2,117 @@
+    <p>
+        В России все хорошо и даже лучше
+    </p>
+
```
Зачем пустые строчки между тегами? Поправь во всем файле

bazhenova commented on index.html:
```diff
@@ -2,7 +2,117 @@
+    </p>
+
+    <div class="image-with-sign">
+        <img src="img.jpg" alt="lemur">
```
Добавь атрибут `title` к картинкам

bazhenova commented on index.css:
```diff
@@ -0,0 +1,114 @@
+@font-face
+{
+    font-family: Overpass;
+    src: url(fonts/Overpass.ttf);
```
Как минимум добавь еще woff формат. Будет вообще круто, если добавишь еще и eot)

bazhenova commented on index.css:
```diff
@@ -0,0 +1,114 @@
+    column-width: 50%;
+    column-rule: 1px solid darkslategray;
+    border-left: 1px solid black;
+    border-top: 1px solid black;
```
Кажется, тут можно обойтись без дублирования кода. Посмотри в сторону `border`, а определенные правила можно задать дополнительно.

bazhenova commented on index.css:
```diff
@@ -0,0 +1,114 @@
+    width: 24%;
+    column-count: 2;
+    column-width: 50%;
+    column-rule: 1px solid darkslategray;
```
Цвета лучше указывать в hex формате. Поправь во всем файле, это не единственное место

bazhenova commented on index.css:
```diff
@@ -0,0 +1,114 @@
+    height: 550px;
+}
+
+.column > p:first-of-type::first-letter 
```
Вместо описывания свойств для определенных тегов, лучше использовать классы. Например, какой-то из первых параграфов ты захочешь оформить по-другому. Тогда можно будет просто навесить на него класс вместо переписывания стилей. Посмотри еще по коду, у тебя есть еще несколько мест, где лучше использовать классы

bazhenova commented on index.css:
```diff
@@ -0,0 +1,114 @@
+{
+    padding-right: 3px;
+    margin-top: 10px;
+    margin-bottom: 10px;
```
Давай будем оптимально записывать свойства. Почитай еще раз про `margin`. Тогда не придется отдельно прописывать свойства `margin-top` и `margin-bottom`

bazhenova commented on index.css:
```diff
@@ -0,0 +1,114 @@
+{
+    display: block;
+    text-align: center;
+    font-size: 200%;
```
Тут тоже бы оптимизировать. Я про `font-size` и `font-family`. Давай вспомним про `font`.

bazhenova commented:
🍅

qoter committed:
Поправил замечания

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

qoter committed:
Удалил лишний шрифт

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

qoter committed:
image width 100%

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

qoter committed:
linter fix

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

qoter committed:
Добавил переносы

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

bazhenova commented:
![image](https://cloud.githubusercontent.com/assets/12410663/20040627/8d7ba5a2-a47c-11e6-873c-efc774413069.png)
Выглядит не очень в Firefox

bazhenova commented on index.css:
```diff
@@ -0,0 +1,105 @@
+@font-face
+{
+    font-family: Edisson;
+    src: url(fonts/Edisson.ttf), url(fonts/Edisson.woff), url(fonts/Edisson.eot);
```
Укажи еще `format` для каждого

bazhenova commented on index.css:
```diff
@@ -0,0 +1,105 @@
+@font-face
+{
+    font-family: Five;
+    src: url(fonts/Five.ttf), url(fonts/Five.woff), url(fonts/Five.eot);
```
`format`

bazhenova commented on index.css:
```diff
@@ -0,0 +1,105 @@
+
+.vertical-text
+{
+    font-size: 120%;
```
`font-*` -> `font`

bazhenova commented on index.css:
```diff
@@ -0,0 +1,105 @@
+    text-indent: 8%;
+}
+
+article.two-columns
```
Можно просто указывать классы без тегов:

```
.two-columns
{
    ...
}
```

bazhenova replied:
И не только в этом месте

bazhenova commented on index.html:
```diff
@@ -2,7 +2,95 @@
+<header>
+    <span class="header-text">Газета узбагозета</span>
+</header>
+
```
Пустые строки все еще здесь)

bazhenova commented on index.css:
```diff
@@ -0,0 +1,105 @@
+
+article.two-columns
+{
+    columns: 2;
```
Безвендорное правило лучше указывать после правил с вендорными префиксами, даже если оно никем не поддерживается

bazhenova commented on index.css:
```diff
@@ -0,0 +1,105 @@
+    -moz-column-rule: 1px solid #2f4f4f;
+}
+
+article
```
Давай вместо тегов лучше будем использовать классы, так как может понадобиться оформить их по-разному. Посмотри во всем файле

bazhenova commented:
Доп. задание будешь делать?

bazhenova commented:
🍅

bazhenova commented on index.css:
```diff
@@ -0,0 +1,105 @@
+.image-in-text
+{
+    width: 100%;
+    height: auto;
```
Тут точно нужно это правило?

qoter replied:
Ну да, иначе картинка будет в полный размер, а так по размеру колонки.

bazhenova replied:
Я имею в виду, что если убрать это правило, то ничего не меняется, у тебя ширина задана
![image](https://cloud.githubusercontent.com/assets/12410663/20047021/e7ca2e52-a4d1-11e6-9062-4e5dbe32bcc8.png)

qoter committed:
Правильный порядок правил.

Исправлен горизонтальный текст.
Указан format для шрифтов.
Удалил пустые строки в html.

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

bazhenova commented:
Еще разок напомню, что можно в css писать просто классы без указания тегов + `height: auto` можно не указывать. А так норм, передаю ментору.

bazhenova commented:
🚀

qoter committed:
Удалил некоторые теги в css.

Удалил height у картинок.

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

mokhov commented on index.css:
```diff
@@ -0,0 +1,104 @@
+
+.first::first-letter
+{
+    font: 400 250% Edisson;
```
а где здесь деградация до дефолтного шрифта?

mokhov commented on index.css:
```diff
@@ -0,0 +1,104 @@
+{
+    display: block;
+    text-align: center;
+    font: normal 200% Edisson;
```
normal будет по-умолчанию

mokhov commented on index.css:
```diff
@@ -0,0 +1,104 @@
+    font-size: 120%;
+    font-weight: 600;
+    display: block;
+    float: left;
```
float запрещен

mokhov commented on index.css:
```diff
@@ -0,0 +1,104 @@
+@font-face
+{
+    font-family: Edisson;
+    src: url(fonts/Edisson.ttf) format('truetype'), url(fonts/Edisson.woff) format('woff'), url(fonts/Edisson.eot) format('embedded-opentype');
```
не хватает local

mokhov commented on index.html:
```diff
@@ -2,7 +2,89 @@
 </head>
 <body>
+<header>
+    <span class="header-text">Газета узбагозета</span>
```
зачем нужен отдельный элемент?

mokhov commented on index.html:
```diff
@@ -2,7 +2,89 @@
+<div class="edition">Издание 31 октября 2016 года</div>
+<main>
+    <article class="two-columns">
+        <p class="first">Пу&shy;тин про&shy;ком&shy;мен&shy;ти&shy;ро&shy;вал фра&shy;зу
```
есть селектор для выбора первого параграфа

mokhov commented:
🍅

qoter committed:
Исправил замечания

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

qoter committed:
fix local

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

qoter commented:
🍏
